File: src/types/backup.ts
export interface BackupCode {
  code: string;
  used: boolean;
  usedAt?: string;
  hashedValue: string;
}

export interface BackupCodesState {
  codes: BackupCode[];
  generatedAt: string;
  lastUsed?: string;
  remainingCodes: number;
}

export interface BackupCodesValidation {
  valid: boolean;
  used: boolean;
  error?: string;
}

export interface BackupCodesGenerationOptions {
  numberOfCodes?: number;
  codeLength?: number;
  format?: 'groups' | 'single';
  groupSize?: number;
} 

File: src/types/backup.py
from dataclasses import dataclass
from typing import List, Optional, Literal


@dataclass
class BackupCode:
    code: str
    used: bool
    usedAt: Optional[str] = None
    hashedValue: str


@dataclass
class BackupCodesState:
    codes: List[BackupCode]
    generatedAt: str
    lastUsed: Optional[str] = None
    remainingCodes: int


@dataclass
class BackupCodesValidation:
    valid: bool
    used: bool
    error: Optional[str] = None


@dataclass
class BackupCodesGenerationOptions:
    numberOfCodes: Optional[int] = None
    codeLength: Optional[int] = None
    format: Optional[Literal['groups', 'single']] = None
    groupSize: Optional[int] = None